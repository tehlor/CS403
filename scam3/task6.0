(define qs (quad 1 0 0 1))
(define is (integrate qs 1))
(define ds (derivate is 1 (stream-car qs)))
(sdisplay 5 qs) (println)
(sdisplay 5 is) (println)
(sdisplay 5 ds) (println)
(inspect (same-stream? qs ds 10 .001))
