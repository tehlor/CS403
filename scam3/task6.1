(define (add-streams s t)
    (cons-stream
        (+ (stream-car s) (stream-car t))
        (add-streams (stream-cdr s) (stream-cdr t))))
(define (scale-stream s f)
    (cons-stream
        (* (stream-car s) f)
        (scale-stream (stream-cdr s) f)))
(define ones (cons-stream 1 ones))
(define ones*f (cons-stream 1 (cons-stream 1 (cons-stream 1 (scale-stream ones 1.1)))))
(inspect (same-stream? ones ones*f 3 .001))
(inspect (same-stream? ones ones*f 5 .001))
(inspect (same-stream? ones ones*f 5 2.0))
(inspect (same-stream? ones ones*f 5 0.2))
(inspect (same-stream? ones ones*f 5 0.1001))
(inspect (same-stream? ones ones*f 5 0.1))
(inspect (same-stream? ones ones*f 5 0.0999))
