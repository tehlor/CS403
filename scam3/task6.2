(define qs (quad -1 -1 -1 0.0142857143))
(define is (integrate qs 0.0142857143))
(define ds (derivate is 0.0142857143 (stream-car qs)))
(sdisplay 5 qs) (println)
(sdisplay 5 is) (println)
(sdisplay 5 ds) (println)
(inspect (same-stream? qs ds 10 .001))
