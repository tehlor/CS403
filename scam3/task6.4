(define qs (quad 1 3 2 0.0909090909))
(define is (integrate qs 0.0909090909))
(define ds (derivate is 0.0909090909 (stream-car qs)))
(sdisplay 5 qs) (println)
(sdisplay 5 is) (println)
(sdisplay 5 ds) (println)
(inspect (same-stream? qs ds 10 .001))
