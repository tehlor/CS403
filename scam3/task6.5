(define qs (quad 1 2 2 0.0125000000))
(define is (integrate qs 0.0125000000))
(define ds (derivate is 0.0125000000 (stream-car qs)))
(sdisplay 5 qs) (println)
(sdisplay 5 is) (println)
(sdisplay 5 ds) (println)
(inspect (same-stream? qs ds 10 .001))
