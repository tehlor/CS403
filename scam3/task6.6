(define qs (quad 1 -3 0 1.0000000000))
(define is (integrate qs 1.0000000000))
(define ds (derivate is 1.0000000000 (stream-car qs)))
(sdisplay 5 qs) (println)
(sdisplay 5 is) (println)
(sdisplay 5 ds) (println)
(inspect (same-stream? qs ds 10 .001))
