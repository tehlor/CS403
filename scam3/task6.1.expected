(same-stream? ones ones*f 3 0.0010000000) is #t
(same-stream? ones ones*f 5 0.0010000000) is #f
(same-stream? ones ones*f 5 2.0000000000) is #t
(same-stream? ones ones*f 5 0.2000000000) is #t
(same-stream? ones ones*f 5 0.1001000000) is #t
(same-stream? ones ones*f 5 0.1000000000) is #f
(same-stream? ones ones*f 5 0.0999000000) is #f
